chainer/datasets/pickle_dataset.py

```diff
@@ -53,11 +53,13 @@ class PickleDataset(dataset_mixin.DatasetMixin):
     Even when a user wants to use a large dataset, this dataset can stores all
     data in a large storage like HDD and each data can be randomly accessible.
 
-    >>> with chainer.datasets.open_pickle_dataset_writer('/path/to/data') as w:
+    >>> import tempfile
+    >>> _, path_to_data = tempfile.mkstemp()
```

Comment:
Could you put these lines into `testsetup::` and add `testcleanup::` to close the file?